spyder/plugins/updatemanager/workers.py

```diff
@@ -0,0 +1,270 @@
+import requests
+from requests.exceptions import ConnectionError, HTTPError, SSLError
+
+# Local imports
+from spyder import __version__
+from spyder.config.base import _, is_stable_version, is_conda_based_app
+from spyder.config.utils import is_anaconda
+from spyder.utils.conda import get_spyder_conda_channel
+from spyder.utils.programs import check_version
+
+# Logger setup
+logger = logging.getLogger(__name__)
+
+CONNECT_ERROR_MSG = _(
+    'Unable to connect to the Spyder update service.'
+    '<br><br>Make sure your connection is working properly.'
+)
+
+HTTP_ERROR_MSG = _(
+    'HTTP error {status_code} when checking for updates.'
+    '<br><br>Make sure your connection is working properly,'
+    'and try again later.'
+)
+
+SSL_ERROR_MSG = _(
+    'SSL certificate verification failed while checking for Spyder updates.'
+    '<br><br>Please contact your network administrator for assistance.'
+)
+
+
+class UpdateDownloadCancelledException(Exception):
+    """Download for installer to update was cancelled."""
+    pass
+
+
+class UpdateDownloadIncompleteError(Exception):
+    """Error occured while downloading file"""
+    pass
+
+
+class WorkerUpdate(QObject):
+    """
+    Worker that checks for releases using either the Anaconda
+    default channels or the Github Releases page without
+    blocking the Spyder user interface, in case of connection
+    issues.
+    """
+    sig_ready = Signal()
+
+    def __init__(self, stable_only):
+        super().__init__()
+        self.stable_only = stable_only
+        self.latest_release = None
+        self.releases = None
+        self.update_available = None
+        self.error = None
+        self.channel = None
+
+    def _check_update_available(self):
+        """Checks if there is an update available from releases."""
+        # Filter releases
+        releases = self.releases.copy()
+        if self.stable_only:
+            # Only use stable releases
+            releases = [r for r in releases if is_stable_version(r)]
+        logger.debug(f"Available versions: {self.releases}")
+
+        self.latest_release = releases[-1] if releases else __version__
+        self.update_available = check_version(__version__,
+                                              self.latest_release, '<')
+
+        logger.debug(f"Update available: {self.update_available}")
+        logger.debug(f"Latest release: {self.latest_release}")
+
+    def start(self):
+        """Main method of the worker."""
+        self.error = None
+        self.latest_release = None
+        self.update_available = False
+        error_msg = None
+        pypi_url = "https://pypi.org/pypi/spyder/json"
+
+        if is_conda_based_app():
+            url = 'https://api.github.com/repos/spyder-ide/spyder/releases'
+        elif is_anaconda():
+            self.channel, channel_url = get_spyder_conda_channel()
+
+            if channel_url is None or self.channel == "pypi":
+                url = pypi_url
```

Comment:
```suggestion
            if channel_url is None:
                return
            elif self.channel == "pypi":
                url = pypi_url
```



Why did you change the previous logic here? If `channel_url` is None, we shouldn't recommend to install Spyder from pip. That means that users have it in development mode or that we don't know how they installed it. So, trying to recommend pip could actually break their envs.